cgra/test/CgraTemplateRTL_test.py

```diff
@@ -77,11 +77,11 @@ def construct(s, DUT, FunctionUnit, FuList, DataType, PredicateType,
                 idTo2d_map)
 
     # Connections
-    s.src_ctrl_pkt.send //= s.dut.recv_from_cpu_ctrl_pkt
-
-    s.dut.send_to_noc.rdy //= 0
-    s.dut.recv_from_noc.val //= 0
-    s.dut.recv_from_noc.msg //= NocPktType(0, 0, 0, 0, 0, 0)
+    # As we always first issue request pkt from CPU to NoC, 
+    # when there is no NoC for single CGRA test, 
+    # we have to connect from_noc and to_noc in testbench.
+    s.src_ctrl_pkt.send //= s.dut.recv_from_cpu_pkt
+    s.dut.recv_from_noc //= s.dut.send_to_noc
```

Comment:
Hi @yo96, this `dut.a //= dut.b` are in the testbench, rather than within the DUT. It works well for pymtl simulation. However, when we do translation and python-verilator co-simulation, I didn't see they are connected, these two pair of ports are never rdy/val, i.e., the connection is mssing.. Can you please confirm my assumption?

Reply:
Hmm, I might have run into similar issues in the past. Maybe try adding a bypass queue in between the two ports?

Reply:
Adding bypass queue inside the test_harness?

Reply:
Right -- so that the signals from the translated block are directly connected.

Reply:
- `dut.a //= dut.b` is also directly connected, why is it different from `BypassQueue`?
 - Would `NormalQueue` also work?